src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseParticipationService.java

```diff
@@ -111,13 +111,18 @@ public TemplateProgrammingExerciseParticipation findTemplateParticipationByProgr
      */
     @Transactional(readOnly = true)
     public ProgrammingExerciseStudentParticipation findStudentParticipationByExerciseIdAndStudentId(Long exerciseId, String username) throws EntityNotFoundException {
-        Optional<ProgrammingExerciseStudentParticipation> participation = studentParticipationRepository.findByExerciseIdAndStudentLogin(exerciseId, username);
-        if (participation.isEmpty()) {
+        Optional<ProgrammingExerciseStudentParticipation> optionalParticipation = studentParticipationRepository.findByExerciseIdAndStudentLogin(exerciseId, username);
+        if (optionalParticipation.isEmpty()) {
             throw new EntityNotFoundException("participation could not be found by exerciseId " + exerciseId + " and user " + username);
         }
+        var participation = optionalParticipation.get();
+        // If the exercise associated with participation is not programming exercise return null
+        if (participation.getProgrammingExercise() == null) {
```

Comment:
I check before to ensure that we don't have uncaught NullPointerException

Reply:
please revert, your test case is wrong if you experience a null pointer exception. Also returning null would lead to other problems here

Reply:
Reverted